libmamba/src/core/transaction.cpp

```diff
@@ -1573,31 +1574,45 @@ namespace mamba
 
         const auto lockfile_data = maybe_lockfile.value();
 
-        constexpr auto default_category = "main";
         constexpr auto default_manager = "conda";
 
-        const auto packages = lockfile_data.get_packages_for(
-            default_category, Context::instance().platform, default_manager);
+        struct
+        {
+            std::vector<PackageInfo> conda, pip;
+        } packages;
+
+        for (const auto& category : categories)
+        {
+            std::vector<PackageInfo> selected_packages;
+
+            selected_packages = lockfile_data.get_packages_for(
+                category, Context::instance().platform, default_manager);
```

Comment:
Let's hardcode "conda" here as well? We're already hardcoding it in the `back_inserter` call.